File: Programmers/Level2/타겟넘버/타겟넘버.js
function dfs(start, val, numbers, target) {
  let cnt = 0;

  if (start === numbers.length) {
    if (val === target) {
      return 1;
    }
  } else {
    cnt += dfs(start + 1, val + numbers[start], numbers, target);
    cnt += dfs(start + 1, val - numbers[start], numbers, target);
  }

  return cnt;
}

function solution(numbers, target) {
  const answer = dfs(0, 0, numbers, target);
  return answer;
}

const numbers = [1, 1, 1, 1, 1];
const target = 3;

console.log(solution(numbers, target));

File: Programmers/Level2/타겟넘버/타겟넘버.py
def solution(numbers, target):

    def dfs(s, val):
        cnt = 0
        if s == len(numbers):
            if val == target:
                return 1
        else:
            cnt += dfs(s + 1, val + numbers[s])
            cnt += dfs(s + 1, val - numbers[s])

        return cnt

    answer = dfs(0, 0)

    return answer


numbers = [1, 1, 1, 1, 1]
target = 3

print(solution(numbers, target))
